adapters/repos/db/lsmkv/cursor_segment_replace.go

```diff
@@ -116,9 +133,13 @@ func (s *segmentCursorReplace) nextWithAllKeys() (segmentReplaceNode, error) {
 }
 
 func (s *segmentCursorReplace) firstWithAllKeys() (segmentReplaceNode, error) {
+	contents := make([]byte, s.segment.dataEndPos)
```

Comment:
entire data part of segment seems to be read here (which might be huge).
`replaceStratParseDataWithKey` can be changed to accept reader, similarly as `ParseCollectionNode`

Reply:
done